app/page.tsx

```diff
@@ -57,6 +60,38 @@ export default function HomePage() {
       }),
   });
 
+  const handleSearch = (query: string) => {
+    const currentQuery = new URLSearchParams(window.location.search);
+    query ? currentQuery.set("s", query) : currentQuery.delete("s");
+    setSearch(query);
+    router.push(`${window.location.pathname}?${currentQuery.toString()}`);
```

Comment:
Can we re write browser history instead of push? Otherwise we will have a big history after doing a search